core/src/test/scala/integration/kafka/api/MetricsTest.scala

```diff
@@ -200,11 +201,15 @@ class MetricsTest extends IntegrationTestHarness with SaslSetup {
 
     // Conversion time less than 1 millisecond is reported as zero, so retry with larger batches until time > 0
     var iteration = 0
+    val retryRecordSize = 2000
     TestUtils.retry(5000) {
       val conversionTimeMs = yammerMetricValue(s"$requestMetricsPrefix,name=MessageConversionsTimeMs,request=Produce").asInstanceOf[Double]
       if (conversionTimeMs <= 0.0) {
-        iteration += 1
-        sendRecords(producers.head, 1000 * iteration, 100, tp)
+        // don't increase iteration if we go over buffer limit
+        val nextBatchSize = (1000 * (iteration + 1)) * retryRecordSize
+        if (nextBatchSize < producerBufferSize)
```

Comment:
Do we need this check at all? We will just be sending more batches without the check and that should be ok?